homeassistant/components/overkiz/cover.py

```diff
@@ -1,40 +1,413 @@
+    ),
+    OverkizCoverDescription(
+        key=UIClass.SWINGING_SHUTTER,
+        device_class=CoverDeviceClass.SHUTTER,
+        current_position_state=OverkizState.CORE_CLOSURE,
+        set_position_command=OverkizCommand.SET_CLOSURE,
+        open_command=OverkizCommand.OPEN,
+        close_command=OverkizCommand.CLOSE,
+        is_closed_fn=is_closed,
+        stop_command=OverkizCommand.STOP,
+    ),
+    # OverkizCoverDescription(
+    #     key=UIClass.VENETIAN_BLIND,
+    #     device_class=CoverDeviceClass.BLIND,
+    #     open_tilt_command="tiltUp",
+    #     close_tilt_command="tiltDown",
+    #     stop_tilt_command=OverkizCommand.STOP
+    # )
+]
+
+SUPPORTED_DEVICES = {description.key: description for description in COVER_DESCRIPTIONS}
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the Overkiz covers from a config entry."""
     data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[OverkizCover] = []
 
-    entities: list[OverkizGenericCover] = [
-        Awning(device.device_url, data.coordinator)
-        for device in data.platforms[Platform.COVER]
-        if device.ui_class == UIClass.AWNING
-    ]
+    for device in data.platforms[Platform.COVER]:
+        if description := SUPPORTED_DEVICES.get(device.widget) or SUPPORTED_DEVICES.get(
+            device.ui_class
+        ):
+            entities.append(
+                OverkizCover(
+                    device.device_url,
+                    data.coordinator,
+                    description,
+                )
+            )
 
-    entities += [
-        VerticalCover(device.device_url, data.coordinator)
-        for device in data.platforms[Platform.COVER]
-        if device.ui_class != UIClass.AWNING
-    ]
+    async_add_entities(entities)
 
-    entities += [
-        LowSpeedCover(device.device_url, data.coordinator)
-        for device in data.platforms[Platform.COVER]
-        if OverkizCommand.SET_CLOSURE_AND_LINEAR_SPEED in device.definition.commands
-    ]
 
-    async_add_entities(entities)
+class OverkizCover(OverkizDescriptiveEntity, CoverEntity):
+    """Representation of an Overkiz Cover."""
+
+    entity_description: OverkizCoverDescription
+
+    def __init__(
+        self,
+        device_url: str,
+        coordinator: OverkizDataUpdateCoordinator,
+        description: EntityDescription,
+    ) -> None:
+        """Initialize the device."""
+        super().__init__(device_url, coordinator, description)
+
+        # Use device url as unique ID for backwards compatibility
+        self._attr_unique_id = self.device.device_url
+
+        # Overkiz does support covers where only tilt commands are supported
+        # and HA sets by default open/close as supported feature which conflicts
+        supported_features = CoverEntityFeature(0)
+
+        if self.entity_description.open_command:
```

Comment:
In the previous code, the supported_features was given by using calls to `self.executor.has_command`  
Isn't it better to depends on that rather than declarative code here?